### PR TITLE
DROOLS-4061 : Guided Score card rules not executed via test scenario.

### PR DESCRIPTION
**Pull request by Rikkola**
https://issues.jboss.org/browse/DROOLS-4061

https://github.com/kiegroup/drools/pull/2356
https://github.com/kiegroup/kie-wb-common/pull/2657
https://github.com/kiegroup/drools-wb/pull/1147

### CODE REVIEW

**Comment by Rikkola** [comment]
Not to myself. Hide the scorecard name selector if the experimental editor is hidden.

**Comment by Rikkola** [comment]
@jomarko , @lanceleverich and all. This adds selector into the test scenario configuration tab. Only insert and assert for the inserted Facts are supported. There is still an issue that needs to be fixed related to broken asserts. I tried few things and could not get any data out of the scorecard run.

**Comment by jomarko** [comment]
Unused dependencies Rikkola#1

File: drools-wb-screens/drools-wb-test-scenario-editor/drools-wb-test-scenario-editor-client/src/main/java/org/drools/workbench/screens/testscenario/client/page/settings/ScorecardSelector.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+                    if (scenario.getModelName() != null) {
+                        view.setSelected(scenario.getModelName());
+                    }
+                }).find(path,
+                        scenario.getPackageName());
+    }
+
+    public void onValueSelected(final String modelName) {
+        if (NONE.equals(modelName)) {
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-test-scenario-editor/drools-wb-test-scenario-editor-client/src/main/java/org/drools/workbench/screens/testscenario/client/page/settings/ScorecardSelector.java]: `Objects.equals(NONE, modelName)` would be more safe.

File: drools-wb-screens/drools-wb-test-scenario-editor/drools-wb-test-scenario-editor-client/src/main/java/org/drools/workbench/screens/testscenario/client/page/settings/ScorecardSelectorViewImpl.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    @Override
+    public void init(final ScorecardSelector presenter) {
+        listBox.setWidth("202px");
+        listBox.addChangeHandler(event -> presenter.onValueSelected(listBox.getSelectedValue()));
+    }
+
+    @Override
+    public void setSelected(final String modelName) {
+        for (int i = 0; i < listBox.getItemCount(); i++) {
+            if (listBox.getItemText(i).equals(modelName)) {
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-test-scenario-editor/drools-wb-test-scenario-editor-client/src/main/java/org/drools/workbench/screens/testscenario/client/page/settings/ScorecardSelectorViewImpl.java]: Could we please use `Objects.equals`?

**Comment by jomarko** [comment]
### Manual review
Passed

File: drools-wb-screens/drools-wb-test-scenario-editor/drools-wb-test-scenario-editor-backend/src/main/java/org/drools/workbench/screens/testscenario/backend/server/ScenarioRunnerService.java [hunk 3]
```diff
@@ -193,6 +193,14 @@ private KieSession loadKSession(KieModule module,
...
                 ksession = sessionService.newKieSession(module,
                                                         ksessionName);
+
+                if (isPMMLScenario(module)) {
+                    /* There is an issue with Scorecards where the first run does not work.
+                    * This is because the tests only passes with clones, this forces the test run to use clones.
+                     */
+                    ksession = sessionService.newKieSession(module,
+                                                            ksessionName);
+                }
```
Comment (Rikkola) [inline drools-wb-screens/drools-wb-test-scenario-editor/drools-wb-test-scenario-editor-backend/src/main/java/org/drools/workbench/screens/testscenario/backend/server/ScenarioRunnerService.java]: Not the clean way to fix this, but the fix that works with the time we have.

File: drools-wb-screens/drools-wb-test-scenario-editor/drools-wb-test-scenario-editor-client/src/main/java/org/drools/workbench/screens/testscenario/client/VerifyFactsPanel.java [hunk 4]
```diff
@@ -61,27 +62,46 @@ public VerifyFactsPanel( FixtureList verifyFacts,
...
         }
-    }
 
+        private Object getName(Fixture fixture) {
+            if (fixture instanceof VerifyFact) {
+                return ((VerifyFact) fixture).getName();
+            } else if (fixture instanceof VerifyScorecardScore) {
+                return TestScenarioConstants.INSTANCE.ScoreCardScore();
+            } else {
+                return "";
```
Comment (danielezonca) [inline drools-wb-screens/drools-wb-test-scenario-editor/drools-wb-test-scenario-editor-client/src/main/java/org/drools/workbench/screens/testscenario/client/VerifyFactsPanel.java]: This branch has been added by this PR, before that the code was `setText("'" + verifyFact.getName() + "'");`. Are you sure about that? If this should not happen I would prefer to throw an exception as `else`. Wdyt?

Comment (Rikkola) [reply drools-wb-screens/drools-wb-test-scenario-editor/drools-wb-test-scenario-editor-client/src/main/java/org/drools/workbench/screens/testscenario/client/VerifyFactsPanel.java]: Yes here exception makes sense. This should always be either of the two types that are in the top if clauses.

**Comment by Rikkola** [comment]
Might have to abandon the idea of getting a full downstream build. Looks like I start getting merge conflicts.

**Pull request merged**